parsers/KR.py

```diff
@@ -0,0 +1,182 @@
+
+        for row in rows:
+            sub_row = row.find("td")
+            generation = sub_row.find("div", {"class": "tdCont alC"})
+
+            # 1,059 style used
+            deformatted_generation = generation.text.replace(",", "")
+
+            try:
+                total += float(deformatted_generation)
+            except ValueError:
+                # NOTE '0 ○' returned when plant shutdown
+                continue
+
+            tag_dt = sub_row.findNext("td")
+            text_dt = tag_dt.find("div", {"class": "tdCont alC"})
+            dt = arrow.get(tag_dt.text, "YYYY-MM-DD HH:mm")
+
+            plant_dts.append(dt)
+
+    dt = timestamp_processor(plant_dts)
+
+    return total, dt
+
+
+def fetch_load(session):
+    req = session.get(LOAD_URL)
+    soup = BeautifulSoup(req.content, 'html.parser')
+
+    current_load = soup.find("th", text = re.compile('Current Load'))
+    load_tag = current_load.findNext("td")
+    load = float(load_tag.text.replace(",", ""))
+
+    tag_dt = soup.find("div", {"class": "date"})
+    dt = arrow.get(tag_dt.text, "(YYYY.MM.DD. HH:mm)")
+
+    return load, dt
+
+
+def fetch_production(zone_key = 'KR', session=None, target_datetime=None, logger=getLogger(__name__)):
+    """
+    Requests the last known production mix (in MW) of a given zone
+    Arguments:
+    zone_key (optional) -- used in case a parser is able to fetch multiple zones
+    session (optional) -- request session passed in order to re-use an existing session
+    target_datetime (optional) -- used if parser can fetch data for a specific day
+    logger (optional) -- handles logging when parser is run as main
+    Return:
+    A dictionary in the form:
+    {
+      'zoneKey': 'FR',
+      'datetime': '2017-01-01T00:00:00Z',
+      'production': {
+          'biomass': 0.0,
+          'coal': 0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': null,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0
+      },
+      'storage': {
+          'hydro': -10.0,
+      },
+      'source': 'mysource.com'
+    }
+    """
+    if target_datetime:
+        raise NotImplementedError('This parser is not yet able to parse past dates')
+
+    s=session or requests.Session()
+
+    hydro, hydro_dt = fetch_hydro(s)
+    nuclear, nuclear_dt = fetch_nuclear(s)
+    load, load_dt = fetch_load(s)
+
+    generation_dts = [hydro_dt, nuclear_dt, load_dt]
+
+    dt_aware = timestamp_processor(generation_dts, with_tz=True, check_delta=True)
+
+    unknown = load - nuclear - hydro
+
+    production = {
+                  'production': {
+                    'nuclear': nuclear,
+                    'hydro': hydro,
+                    'unknown': unknown
+                  },
+                  'source': 'khnp.co.kr, kpx.or.kr',
+                  'zoneKey': zone_key,
+                  'datetime': dt_aware.datetime,
+                  'storage': {}
```

Comment:
better just remove that line if you don't plan on putting storage

Reply:
actually according to Alex there's pumped storage and fuel cells (battery?) so I'll add them